lib/eventsource.js

```diff
@@ -10,6 +10,18 @@ var httpsOptions = [
   'rejectUnauthorized', 'secureProtocol', 'servername'
 ]
 
+var bom = [239, 187, 191]
+var colon = 58
+var space = 32
+var lineFeed = 10
+var carriageReturn = 13
+
+function hasBom (buf) {
+  return bom.every(function (charCode, index) {
+    return buf[index] === charCode
+  })
```

Comment:
slice + compare Buffer functions

Reply:
Constructing a buffer for the BOM would require one of the following:
- Using `Buffer.from()` which isn't present in all node versions we currently support
- Using `new Buffer()` which is deprecated in newer versions of node
- Adding and using `buffer-from` module (introducing a dependency)
- Replicate logic from `buffer-from` module, introducing node version checks in code

Performance-wise, we are only doing this for the start of the stream, so shouldn't matter. If we bump the minimum version to node 6 for this module and bump the version to 2.x, we could introduce a bunch of similar  changes, but for now I think this is an acceptable fix without breaking compatibility with older versions of node.